packages/snaps-simulator/src/components/Icon.tsx

```diff
@@ -192,14 +198,24 @@ export const Icon: ForwardRefExoticComponent<IconProps> = forwardRef(
       ...props
     },
     ref,
-  ) => (
-    <Image
-      ref={ref}
-      src={DEFAULT_ICONS[icon].src}
-      alt={alt}
-      width={width}
-      height={height}
-      {...props}
-    />
-  ),
+  ) => {
+    const { colorMode } = useColorMode();
+
+    const iconMetadata = DEFAULT_ICONS[icon];
+    const defaultSrc = iconMetadata.src;
+    const darkSrc = (iconMetadata as any).srcDark ?? iconMetadata.src;
```

Comment:
Can we use `hasProperty` instead of a type cast?

Reply:
Done!